cmd/operator-sdk/cli/cli.go

```diff
@@ -23,7 +23,7 @@ import (
 	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/olm"
 	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/version"
 	"github.com/operator-framework/operator-sdk/internal/flags"
-	"github.com/operator-framework/operator-sdk/internal/plugins/golang"
+	golangv2 "github.com/operator-framework/operator-sdk/internal/plugins/golang/v2"
```

Comment:
Could we not just start from v3-alpha instead of having here v2 and v3 too?

Reply:
IMO v2-alpha makes sense for the SDK plugin. I'm thinking of the go.operator-sdk.io/v1 plugin as the legacy scaffold code and go.operator-sdk.io/v2-alpha as the new project layout plugin, similar to how kubebuilder used plugins go/v1 and go/v2 for project versions 1 and 2, then created a plugin go/v3-alpha for project version 3-alpha (disregard version similarities).

Reply:
I think I could not properly clarify my suggestion. So, I mean: 

Instead of we move with this PR and end up with: (supporting both and soon deprecating v2)

├── v2
│   ├── api.go
│   ├── config.go
│   ├── init.go
│   └── plugin.go
├── v3
│   ├── api.go
│   ├── config.go
│   ├── init.go
│   └── plugin.go
└── plugins.go
 
IMO we could just keep it as it is: 

├── golang
│   ├── api.go
│   ├── config.go
│   ├── init.go
│   └── plugin.go
└── plugins.go

And then point to v3 projects and v3-alpha plugin (which I understand that would be the default for V3 projects versions) since it is the version where we actually are using and scaffolding. See that the new layout for SDK is using PROJECT version v3-alpha. The plugins are not applied in the PROJECT for v2. 
 
```yaml
...
version: 3-alpha
plugins:
  go.operator-sdk.io/v2.0.0: {}
```

Then, the PR for SDK would be like the draft made to test it: https://github.com/operator-framework/operator-sdk/pull/3150/files

Reply:
I don't think the approach you suggest is correct. We should mirror how plugins are packaged in kubebuilder, which is no longer based on project version but plugin version. See https://github.com/kubernetes-sigs/kubebuilder/pull/1551 for an example.

Reply:
I think you are right.